### PR TITLE
RFC: Code of Conduct and updated Issue/PR templates

### PR DESCRIPTION
**Pull request by aem**
Here's a proposal for a Code of Conduct (with pieces of both the WeAllJS and Facebook templates), and PR/Issue templates (a combination of the existing content and the React templates).

Any feedback is welcome!

cc/ @bvaughn @diogofcunha

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -4,15 +4,28 @@ Please don't file GitHub issues to ask questions! Instead use:
...
+
+-->
+
+**What is the current behavior?**
+
+**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have dependencies other than React. Paste the link to your JSFiddle (https://jsfiddle.net/Luktwrdm/) or CodeSandbox (https://codesandbox.io/s/new) example below:**
+
+**What is the expected behavior?**
+
+**Which versions of react-virtualized, and which browser / OS are affected by this issue? Did this work in previous versions of React?**
```
Comment (diogofcunha) [inline .github/ISSUE_TEMPLATE.md]: https://github.com/bvaughn/react-virtualized/blob/master/package.json#L162 probably also want some info about react version as we support multiple?

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -4,15 +4,28 @@ Please don't file GitHub issues to ask questions! Instead use:
...
+
+Please include either a failing unit test or a simple repro. You can start by forking one of these:
+* Code Sandbox: https://codesandbox.io/s/03qpzq1p9p?module=%2FExample.js
+* Plnkr: https://plnkr.co/edit/6syKo8cx3RfoO96hXFT1
+
+-->
+
+**What is the current behavior?**
+
+**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have dependencies other than React. Paste the link to your JSFiddle (https://jsfiddle.net/Luktwrdm/) or CodeSandbox (https://codesandbox.io/s/new) example below:**
```
Comment (diogofcunha) [inline .github/ISSUE_TEMPLATE.md]: These are the react code boxes, we should probably replace this with the ones we already had for this project?

https://codesandbox.io/s/03qpzq1p9p?module=%2FExample.js
https://plnkr.co/edit/6syKo8cx3RfoO96hXFT1

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -4,15 +4,28 @@ Please don't file GitHub issues to ask questions! Instead use:
...
+
+Reporting a Bug?
+
+Please include either a failing unit test or a simple repro. You can start by forking one of these:
+* Code Sandbox: https://codesandbox.io/s/03qpzq1p9p?module=%2FExample.js
+* Plnkr: https://plnkr.co/edit/6syKo8cx3RfoO96hXFT1
+
+-->
+
+**What is the current behavior?**
```
Comment (diogofcunha) [inline .github/ISSUE_TEMPLATE.md]: We should probably highlight the section splits with h3 or h2 to make a noticeable difference with the descriptions, thoughts?

File: .github/ISSUE_TEMPLATE.md (hunk 1 above)
Comment (diogofcunha) [reply .github/ISSUE_TEMPLATE.md]: Maybe "previous versions of r-v?"

File: .github/ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -4,15 +4,28 @@ Please don't file GitHub issues to ask questions! Instead use:
...
 
-# Reporting a Bug?
+**Do you want to request a *feature* or report a *bug*?**
 
-Please include either a failing unit test or a simple repro. You can start by forking one of these:
-* Code Sandbox: https://codesandbox.io/s/03qpzq1p9p?module=%2FExample.js
-* Plnkr: https://plnkr.co/edit/6syKo8cx3RfoO96hXFT1
-
-# Requesting a Feature?
+<!--
```
Comment (diogofcunha) [inline .github/ISSUE_TEMPLATE.md]: Shall we just delete instead of comment?

Comment (bvaughn) [reply .github/ISSUE_TEMPLATE.md]: I'm not sure I understand the use of the comment  tag

Even though these are markdown files, when you open a new issue- they're going to appear in plaintext. (See [here](https://github.com/bvaughn/react-virtualized/issues/new) for an example)

So the comment tag won't actually work, nor will any of the fancier formatting :)

Comment (aem) [reply .github/ISSUE_TEMPLATE.md]: @bvaughn that's intentional. i think those are things we want people to think about, but text that we don't want showing up in the final issue that's opened, so if it's there as a comment we don't have to depend on people to delete the text. should keep the same intent of the original content without cluttering up issue descriptions.

Comment (bvaughn) [reply .github/ISSUE_TEMPLATE.md]: I see. I would expect people to delete the contents of the issue template (and replace with their own issue description, etc.)

It doesn't make much sense to leave the template content in the issue, I think.

Comment (aem) [reply .github/ISSUE_TEMPLATE.md]: ok! happy to uncomment and just leave it there as guiding text

**Comment by aem** [comment]
everyone's had some time to think this one over. are we feeling good about where these are at?

File: .github/ISSUE_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+Reporting a Bug?
+
+Please include either a failing unit test or a simple repro. You can start by forking one of these:
+
+* Code Sandbox: https://codesandbox.io/s/03qpzq1p9p?module=%2FExample.js
+* Plnkr: https://plnkr.co/edit/6syKo8cx3RfoO96hXFT1
+
+### What is the current behavior?
+
+If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have dependencies other than React and react-virtualized. Paste the link to your JSFiddle (https://codesandbox.io/s/03qpzq1p9p?module=%2FExample.js) or Plnkr (https://plnkr.co/edit/6syKo8cx3RfoO96hXFT1) example below:
```
Comment (wuweiweiwu) [inline .github/ISSUE_TEMPLATE.md]: jsfiddle, plnkr or codesandbox? 

Can we keep it to codesandbox? its easiest for maintainers.

File: .github/ISSUE_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+### What is the expected behavior?
+
+### Which versions of React and react-virtualized, and which browser / OS are affected by this issue? Did this work in previous versions of react-virtualized?
+|                   |          |
+|-------------------|----------|
+| Browser           |          |
+| OS                |          |
+| React             |          |
+| react-virtualized |          |
```
Comment (wuweiweiwu) [inline .github/ISSUE_TEMPLATE.md]: nit: add react-dom

File: .github/ISSUE_TEMPLATE.md (hunk 5 above)
Comment (aem) [reply .github/ISSUE_TEMPLATE.md]: 👍  some of this was just a merge of the existing templates and the React templates, i'll pare it down to just codesandbox

**Comment by wuweiweiwu** [comment]
Looks awesome!

**Pull request merged**